zerver/lib/narrow.py

```diff
@@ -23,7 +23,9 @@ def check_supported_events_narrow_filter(narrow: Iterable[Sequence[str]]) -> Non
         if operator not in ["stream", "topic", "sender", "is"]:
             raise JsonableError(_("Operator {} not supported.").format(operator))
 
-def is_web_public_compatible(narrow: Iterable[Dict[str, str]]) -> bool:
+def is_web_public_compatible(narrow: Optional[Iterable[Dict[str, Any]]]) -> bool:
+    assert narrow is not None
```

Comment:
Why is the type of this `Optional` and then right away we assert not None? Probably worth a comment explaining it, if it's intentional

Reply:
Hmm, shouldn't this take an `OptionalNarrowListT`?  Or maybe better would be to have it take a non-optional `NarrowListT` and have the calling code do this:
```
        if not is_web_public_narrow(narrow):                                                                             
            return json_unauthorized()                                                                                   
        assert narrow is not None                                                                                        
        if not is_web_public_compatible(narrow):                                                                         
            return json_unauthorized()                                                                                   
```

Reply:
Okay! updated!